Cleanup and Refactoring

- Fix eslint warnings
./node_modules/eslint/bin/eslint.js --config .eslintrc .

```
/Users/abhay/Documents/projects/branded-token.js/test/utils/configReader.js
  23:22  warning  Unexpected unnamed function  func-names

/Users/abhay/Documents/projects/branded-token.js/test/utils/helper.js
  23:20  warning  Unexpected unnamed function  func-names

✖ 2 problems (0 errors, 2 warnings)
```

- Update with missing changelog.

- Evaluate if anything needs to be updated in readme and update accordingly.

- Move hardcoded constants of integration tests to configReader.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,7 +2,9 @@
 
 ## Version 0.10.0
 
-* Illustrate reject stakeRequest flow in integration test ([#62] (https://github.com/OpenSTFoundation/brandedtoken.js/pull/62))
+* Readme ([#72](https://github.com/OpenSTFoundation/brandedtoken.js/pull/72))
+* LICENSE Update and repo alignment as per eslint configuration ([#71](https://github.com/OpenSTFoundation/brandedtoken.js/pull/71))
+* Illustrate reject stakeRequest flow in integration test ([#62](https://github.com/OpenSTFoundation/brandedtoken.js/pull/62))
 * Updated package.json and npm run alignment ([#65](https://github.com/OpenSTFoundation/brandedtoken.js/pull/65))
 * Integration test files cleanup ([#60](https://github.com/OpenSTFoundation/brandedtoken.js/pull/60))
 * Use docker geth instance for testing ([#58](https://github.com/OpenSTFoundation/brandedtoken.js/pull/58))

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,8 +4,8 @@
 
 * Replace mosaic-tbd with mosaic.js ([#74](https://github.com/OpenSTFoundation/brandedtoken.js/pull/74))
 * Readme ([#72](https://github.com/OpenSTFoundation/brandedtoken.js/pull/72))
-* LICENSE Update and repo alignment as per eslint configuration ([#71](https://github.com/OpenSTFoundation/brandedtoken.js/pull/71))
-* Illustrate reject stakeRequest flow in integration test ([#62] (https://github.com/OpenSTFoundation/brandedtoken.js/pull/62))
+* License Update and repo alignment as per eslint configuration ([#71](https://github.com/OpenSTFoundation/brandedtoken.js/pull/71))
+* Illustrate reject stakeRequest flow in integration test ([#62](https://github.com/OpenSTFoundation/brandedtoken.js/pull/62))
 * Updated package.json and npm run alignment ([#65](https://github.com/OpenSTFoundation/brandedtoken.js/pull/65))
 * Integration test files cleanup ([#60](https://github.com/OpenSTFoundation/brandedtoken.js/pull/60))
 * Use docker geth instance for testing ([#58](https://github.com/OpenSTFoundation/brandedtoken.js/pull/58))

--- a/README.MD
+++ b/README.MD
@@ -207,14 +207,16 @@ mockContractDeployerObject.deployMockGatewayPass().then(function(){
 ## BrandedToken Staking through GatewayComposer
 
 ### Approve GatewayComposer for ValueToken and call requestStake
+Please make sure staker has sufficient stake amount before calling this step.
 
 ```js
 const StakeHelper = BrandedToken.Helpers.StakeHelper;
 const stakeHelperInstance = new StakeHelper(web3Provider, brandedTokenAddress, gatewayComposerAddress);
 let txOptions = {
-  from: staker
+  from: staker,
+  gas: '7500000',
 };
-const stakeAmountInWei = '1000000000000';
+const stakeAmountInWei = '10000000000';
 
 // stakerGatewayNonce is the nonce of the staker address stored in Gateway contract. Alternatively it can also be fetched from Gateway.getNonce(stakeAddress) method.
 const stakerGatewayNonce = 1;
@@ -272,6 +274,7 @@ gatewayContractInstance.methods.bounty().call().then(function(result){
 const Facilitator = BrandedToken.Helpers.Facilitator;
 const facilitatorInstance = new Facilitator(web3Provider, eip20Token, brandedTokenAddress, gatewayComposerAddress, facilitator);
 txOptions = {
+  from: facilitator,
   gas: '7500000'
 };    
 let stakeRequestHash; // stakeRequestHash unique per stake request stored in GatewayComposer.stakeRequests mapping

--- a/lib/helpers/stake/gateway_composer/Facilitator.js
+++ b/lib/helpers/stake/gateway_composer/Facilitator.js
@@ -85,25 +85,24 @@ class Facilitator {
       oThis.brandedToken,
       oThis.gatewayComposer,
     );
-    await stakeHelperInstance.approveForBounty(
+    const approveForBountyResult = await stakeHelperInstance.approveForBounty(
       oThis.facilitator,
       bountyInWei,
       oThis.valueToken,
       valueTokenAbi,
       originWeb3,
-    ).then((receipt) => {
-      console.log('receipt for approveForBounty', receipt);
-    });
-    await stakeHelperInstance.acceptStakeRequest(
+    );
+    console.log('approveForBounty result:', approveForBountyResult);
+
+    const acceptStakeRequestResult = await stakeHelperInstance.acceptStakeRequest(
       stakeRequestHash,
       signature,
       oThis.facilitator,
       hashLock,
       oThis.originWeb3,
       txOptions,
-    ).then((receipt) => {
-      console.log('receipt for acceptStakeRequest', receipt);
-    });
+    );
+    console.log('acceptStakeRequest result:', acceptStakeRequestResult);
   }
 }
 

--- a/lib/helpers/stake/gateway_composer/Facilitator.js
+++ b/lib/helpers/stake/gateway_composer/Facilitator.js
@@ -92,7 +92,7 @@ class Facilitator {
       valueTokenAbi,
       originWeb3,
     );
-    console.log('approveForBounty result:', approveForBountyResult);
+    console.log('approveForBounty status:', approveForBountyResult.status);
 
     const acceptStakeRequestResult = await stakeHelperInstance.acceptStakeRequest(
       stakeRequestHash,
@@ -102,7 +102,7 @@ class Facilitator {
       oThis.originWeb3,
       txOptions,
     );
-    console.log('acceptStakeRequest result:', acceptStakeRequestResult);
+    console.log('acceptStakeRequest status:', acceptStakeRequestResult.status);
   }
 }
 

--- a/lib/helpers/stake/gateway_composer/Staker.js
+++ b/lib/helpers/stake/gateway_composer/Staker.js
@@ -81,16 +81,17 @@ class Staker {
       oThis.brandedToken,
       oThis.gatewayComposer,
     );
-    await stakeHelperInstance.approveForValueToken(
+    const approveForValueTokenResult = await stakeHelperInstance.approveForValueToken(
       oThis.valueToken,
       valueTokenAbi,
       stakeVTAmountInWei,
       oThis.originWeb3,
       txOptions,
-    ).then((receipt) => {
-      console.log('receipt for approveForValueToken', receipt);
-    });
-    await stakeHelperInstance.requestStake(
+    );
+
+    console.log('approveForValueToken result:', approveForValueTokenResult);
+
+    const requestStakeResult = await stakeHelperInstance.requestStake(
       owner,
       stakeVTAmountInWei,
       mintBTAmountInWei,
@@ -101,9 +102,9 @@ class Staker {
       stakerGatewayNonce,
       oThis.originWeb3,
       txOptions,
-    ).then((receipt) => {
-      console.log('receipt for requestStake', receipt);
-    });
+    );
+
+    console.log('requestStake result:', requestStakeResult);
   }
 }
 

--- a/lib/helpers/stake/gateway_composer/Staker.js
+++ b/lib/helpers/stake/gateway_composer/Staker.js
@@ -89,7 +89,7 @@ class Staker {
       txOptions,
     );
 
-    console.log('approveForValueToken result:', approveForValueTokenResult);
+    console.log('approveForValueToken status:', approveForValueTokenResult.status);
 
     const requestStakeResult = await stakeHelperInstance.requestStake(
       owner,
@@ -104,7 +104,7 @@ class Staker {
       txOptions,
     );
 
-    console.log('requestStake result:', requestStakeResult);
+    console.log('requestStake status:', requestStakeResult.status);
   }
 }
 

--- a/test/integration/BrandedToken/RejectStakeRequest.js
+++ b/test/integration/BrandedToken/RejectStakeRequest.js
@@ -25,7 +25,7 @@ const BN = require('bn.js');
 const chai = require('chai');
 
 const Web3 = require('web3');
-const Mosaic = require('@openstfoundation/mosaic-tbd');
+const Mosaic = require('@openstfoundation/mosaic.js');
 const Package = require('../../../index');
 
 const Setup = Package.EconomySetup;

--- a/test/integration/BrandedToken/RejectStakeRequest.js
+++ b/test/integration/BrandedToken/RejectStakeRequest.js
@@ -21,10 +21,11 @@
 'use strict';
 
 // Load external packages
+const BN = require('bn.js');
 const chai = require('chai');
 
 const Web3 = require('web3');
-
+const Mosaic = require('@openstfoundation/mosaic-tbd');
 const Package = require('../../../index');
 
 const Setup = Package.EconomySetup;
@@ -65,6 +66,8 @@ describe('RejectStakeRequest', async () => {
     originWeb3 = new Web3(rpcEndpointOrigin);
     accountsOrigin = await originWeb3.eth.getAccounts();
     [deployerAddress, beneficiary] = accountsOrigin;
+    // Deployer while deploying MockToken gets MAX ValueTokens.
+    // Since owner is the deployer, owner also gets MAX ValueTokens.
     owner = deployerAddress;
   });
 
@@ -84,7 +87,7 @@ describe('RejectStakeRequest', async () => {
       deployer: deployerAddress,
       owner,
       workers: worker,
-      workerExpirationHeight: '20000000',
+      workerExpirationHeight: config.workerExpirationHeight,
     };
     await orgHelper.setup(orgConfig);
     caOrganization = orgHelper.address;
@@ -102,11 +105,11 @@ describe('RejectStakeRequest', async () => {
     const btHelperConfig = {
       deployer: deployerAddress,
       valueToken: caMockToken,
-      symbol: 'BT',
-      name: 'MyBrandedToken',
-      decimals: '18',
-      conversionRate: '1000',
-      conversionRateDecimals: 5,
+      symbol: config.symbol,
+      name: config.name,
+      decimals: config.decimals,
+      conversionRate: config.conversionRate,
+      conversionRateDecimals: config.conversionRateDecimals,
       organization: caOrganization,
     };
 
@@ -148,6 +151,17 @@ describe('RejectStakeRequest', async () => {
     await deployerInstance.deployMockGatewayPass();
     caGateway = deployerInstance.addresses.MockGatewayPass;
     assert.isNotNull(caGateway, 'Gateway contract address should not be null.');
+  });
+
+  it('Verifies ValueToken balance of owner', async () => {
+    const mockTokenInstance = Mosaic.Contracts.getEIP20Token(
+      originWeb3,
+      caMockToken,
+    );
+    const balanceOfStaker = await mockTokenInstance.methods.balanceOf(owner).call();
+    const balanceOfStakerBN = new BN(balanceOfStaker);
+    const stakeAmountBN = new BN(config.stakeAmountInWei);
+    assert.strictEqual(balanceOfStakerBN.cmp(stakeAmountBN), 1, 'staker ValueToken balance should be greater/equal to stakeAmountInWei.');
   });
 
   it('Performs staker.requestStake', async () => {

--- a/test/integration/GatewayComposer/BrandedTokenStake.js
+++ b/test/integration/GatewayComposer/BrandedTokenStake.js
@@ -20,8 +20,8 @@
 
 'use strict';
 
-
 // Load external packages
+const BN = require('bn.js');
 const chai = require('chai');
 const Web3 = require('web3');
 const Mosaic = require('@openstfoundation/mosaic-tbd');
@@ -67,9 +67,9 @@ describe('Performs BrandedToken staking through GatewayComposer', async () => {
     originWeb3 = new Web3(rpcEndpointOrigin);
     const accountsOrigin = await originWeb3.eth.getAccounts();
     [deployerAddress, facilitator, beneficiary] = accountsOrigin;
+    // Deployer while deploying MockToken gets MAX ValueTokens.
+    // Since owner is the deployer, owner also gets MAX ValueTokens.
     owner = deployerAddress;
-    // [facilitator] = accountsOrigin[1];
-    // [beneficiary] = accountsOrigin[2];
   });
 
   after(() => {
@@ -86,7 +86,7 @@ describe('Performs BrandedToken staking through GatewayComposer', async () => {
       deployer: deployerAddress,
       owner,
       workers: worker,
-      workerExpirationHeight: '20000000',
+      workerExpirationHeight: config.workerExpirationHeight,
     };
     await orgHelper.setup(orgConfig);
     caOrganization = orgHelper.address;
@@ -104,11 +104,11 @@ describe('Performs BrandedToken staking through GatewayComposer', async () => {
     const btHelperConfig = {
       deployer: deployerAddress,
       valueToken: caMockToken,
-      symbol: 'BT',
-      name: 'MyBrandedToken',
-      decimals: '18',
-      conversionRate: '1000',
-      conversionRateDecimals: 5,
+      symbol: config.symbol,
+      name: config.name,
+      decimals: config.decimals,
+      conversionRate: config.conversionRate,
+      conversionRateDecimals: config.conversionRateDecimals,
       organization: caOrganization,
     };
 
@@ -138,7 +138,6 @@ describe('Performs BrandedToken staking through GatewayComposer', async () => {
 
     const gcHelper = new GatewayComposerHelper(originWeb3, null);
 
-
     const gatewayComposerInstance = await gcHelper.setup(gcHelperConfig, gcDeployParams);
 
     gatewayComposerAddress = gatewayComposerInstance.contractAddress;
@@ -150,6 +149,17 @@ describe('Performs BrandedToken staking through GatewayComposer', async () => {
     await deployerInstance.deployMockGatewayPass();
     caGateway = deployerInstance.addresses.MockGatewayPass;
     assert.isNotNull(caGateway, 'Gateway contract address should not be null.');
+  });
+
+  it('Verifies ValueToken balance of owner', async () => {
+    const mockTokenInstance = Mosaic.Contracts.getEIP20Token(
+      originWeb3,
+      caMockToken,
+    );
+    const balanceOfStaker = await mockTokenInstance.methods.balanceOf(owner).call();
+    const balanceOfStakerBN = new BN(balanceOfStaker);
+    const stakeAmountBN = new BN(config.stakeAmountInWei);
+    assert.strictEqual(balanceOfStakerBN.cmp(stakeAmountBN), 1, 'staker ValueToken balance should be greater/equal to stakeAmountInWei.');
   });
 
   it('Performs staker.requestStake', async () => {
@@ -166,7 +176,6 @@ describe('Performs BrandedToken staking through GatewayComposer', async () => {
       originWeb3,
       txOptions,
     );
-
 
     const stakerGatewayNonce = 1;
 
@@ -195,7 +204,8 @@ describe('Performs BrandedToken staking through GatewayComposer', async () => {
       originWeb3,
       txOptions,
     );
-    assert.strictEqual(gatewayComposerAddress, btStakeStruct.staker, 'Incorrect staker address');
+    assert.strictEqual(gatewayComposerAddress, btStakeStruct.staker, 'Incorrect staker address.');
+    assert.strictEqual(config.stakeAmountInWei, btStakeStruct.stake, 'Incorrect stake amount.');
   });
 
   it('Validates worker is whitelisted', async () => {

--- a/test/utils/configReader.js
+++ b/test/utils/configReader.js
@@ -20,7 +20,7 @@
 
 'use strict';
 
-const ConfigReader = function () {};
+const ConfigReader = function constructor() {};
 
 ConfigReader.prototype = {
   gasPrice: '0x3B9ACA00',

--- a/test/utils/configReader.js
+++ b/test/utils/configReader.js
@@ -27,9 +27,15 @@ ConfigReader.prototype = {
   gas: 7500000,
   nullBytes32: '0x0000000000000000000000000000000000000000000000000000000000000000',
   originPort: 8546,
-  stakeAmountInWei: '200',
+  stakeAmountInWei: '10000000000',
   stakeGasPrice: '7500000',
   stakeGasLimit: '100',
+  workerExpirationHeight: '20000000',
+  symbol: 'BT',
+  name: 'MyBrandedToken',
+  decimals: '18',
+  conversionRate: '10',
+  conversionRateDecimals: '5',
 };
 
 module.exports = new ConfigReader();

--- a/test/utils/helper.js
+++ b/test/utils/helper.js
@@ -20,7 +20,7 @@
 
 'use strict';
 
-const TestHelper = function (homeDir) {
+const TestHelper = function constructor(homeDir) {
   const oThis = this;
 
   oThis.homeDir = homeDir;

